### ISSUE TITLE
Accessibility: Auto-generated IDs can start with a "-"

### ISSUE THREAD

**Issue by NickDJM** [NickDJM/accessible-menu#87]
## Summary

What an ID is generated through a `MenuToggle`, it can lead to accessibility violations.

If the `innerText` of a toggle starts with a space is it replaced with a `-`, which is technically an accessibility violation. The same issue can also cause _many_ dashes to be added to the IDs if there are spacing issues with the DOM.

Another side-effect of the current way IDs are generated is if a developer is using `aria-label` instead of `innerText` nothing is added to the ID string, which also causes it to start with a dash since the the string is essentially: `[innerText]-[randomString]-[menu/menu-button]`.

## Expected Behaviour

I think the best way to go about fixing this will be to do a couple things:
1. If there is no `innerText`, see if there's an `aria-label` that can be used,
2. Replace _any_ amount of spaces with a single `-`,
3. Make sure the string generated by `innerText`/`aria-label` does not start or end with a `-`, and
4. Make the final generated ID check to see if the generated string is empty before adding it to the beginning of the rest of the ID. Should fall-back to just `[randomString]-[menu/menu-button]`

## Screenshots

You can see the dash being added here because there is no `innerText`

```html
<button class="navbar-toggler" type="button" aria-expanded="false" aria-label="Toggle navigation" aria-haspopup="true" id="-qefmxrqg-menu-button" aria-controls="-qefmxrqg-menu">
  <span class="navbar-toggler-icon"></span>
</button>
```

Here's a Bootstrap 3/Drupal example where the generated markup is causing a _load_ of extras dashes.

```html
<button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#navbar-collapse" aria-haspopup="true" aria-expanded="false" role="button" id="-------------toggle-navigation---------------------------------------------------uikbq-menu-button" aria-controls="-------------toggle-navigation---------------------------------------------------uikbq-menu">
            <span class="sr-only">Toggle navigation</span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
          </button>
```

## Spec

- https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id
